src/sysfs.rs

```diff
@@ -177,4 +189,29 @@ impl GpioIn for SysFsGpioInput {
             }
         }
     }
+
+    fn set_edge(&mut self, edge: GpioEdge) -> Result<(), Self::Error> {
+        fs::OpenOptions::new()
+            .write(true)
+            .open(format!("/sys/class/gpio/gpio{}/edge", self.gpio.gpio_num))?
+            .write_all(match edge {
+                GpioEdge::None => b"none",
+                GpioEdge::Rising => b"rising",
+                GpioEdge::Falling => b"falling",
+                GpioEdge::Both => b"both",
+            })?;
+        Ok(())
+    }
+
+    fn wait_for_edge(&mut self, timeout_ms: u64) -> Result<Option<GpioValue>, Self::Error> {
+        let _ = self.read_value()?;
```

Comment:
What's this for precisely? Does epoll require us to read once first?

Reply:
I think that it would otherwise fire immediately if there had been an edge since the previous read, but I'm actually not sure. I'll try it out.

And even if so: Maybe that would actually be the desired behavior?

Reply:
I tried it out: Removing that read does indeed make the epoll return immediately if there was a corresponding edge in the past. E.g. I saw a poll for _rising_ edges return with value `0`, just because it had gone from `0` to `1` and back before `wait_for_edge` was called.